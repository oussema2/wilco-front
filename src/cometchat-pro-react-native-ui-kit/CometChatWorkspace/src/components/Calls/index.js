export { default as CometChatIncomingCall } from './CometChatIncomingCall';
export { default as CometChatOutgoingCall } from './CometChatOutgoingCall';
export { default as CometChatOutgoingDirectCall } from './CometChatOutgoingDirectCall';
export { default as CometChatIncomingDirectCall } from './CometChatIncomingDirectCall';
