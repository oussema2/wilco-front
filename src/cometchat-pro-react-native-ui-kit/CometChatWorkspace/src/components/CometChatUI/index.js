/* eslint-disable import/no-duplicates */
import React, { useEffect, useRef, useState } from 'react';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { View, Text } from 'react-native';
import { CometChatGroupListWithMessages } from '../Groups';
import { CometChatUserListWithMessages } from '../Users';
import { CometChatConversationListWithMessages } from '../Chats';
import { CometChatContextProvider } from '../../utils/CometChatContext';
import { CometChatUserProfile } from '../UserProfile';
import Ionicons from 'react-native-vector-icons/Ionicons';
import MaterialIcons from 'react-native-vector-icons/MaterialIcons';
import MCIIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import theme from '../../resources/theme';
import { heightRatio } from '../../utils/consts';


const Tab = createBottomTabNavigator();

function CometChatUI() {
  const [tabs, setTabs] = useState(null);
  const contextRef = useRef(null);

  useEffect(() => {
    checkRestrictions();
  }, []);
  const checkRestrictions = async () => {
    let isChatEnabled = await contextRef.current.state.FeatureRestriction.isRecentChatListEnabled();
    let isGroupListEnabled = await contextRef.current.state.FeatureRestriction.isGroupListEnabled();
    let isUserSettingsEnabled = await contextRef.current.state.FeatureRestriction.isUserSettingsEnabled();
    let isUserListEnabled = await contextRef.current.state.FeatureRestriction.isUserListEnabled();
    let isCallListEnabled = await contextRef.current.state.FeatureRestriction.isCallListEnabled();
    setTabs({
      isChatEnabled,
      isGroupListEnabled,
      isUserSettingsEnabled,
      isUserListEnabled,
      isCallListEnabled,
    });
  };
  return (
    <CometChatContextProvider ref={contextRef}>
      {tabs ? (
        <Tab.Navigator
          screenOptions={({ route }) => ({
            tabBarIcon: ({ color }) => {
              let iconName;

              if (route.name === 'Chats') {
                return (
                  <MCIIcons name="chat" size={25 * heightRatio} color={color} />
                );
              }
              if (route.name === 'More') {
                return (
                  <MaterialIcons
                    name="more-horiz"
                    size={25 * heightRatio}
                    color={color}
                  />
                );
              }
              if (route.name === 'Users') {
                iconName = 'ios-person-circle-sharp';
              } else if (route.name === 'Groups') {
                iconName = 'people';
              }

              // You can return any component that you like here!
              return (
                <Ionicons
                  name={iconName}
                  size={24 * heightRatio}
                  color={color}
                />
              );
            },
          })}
          tabBarOptions={{
            activeTintColor: theme.color.blue,
            inactiveTintColor: 'rgba(0,0,0,0.5)',
            activeBackgroundColor: theme.color.white,
            inactiveBackgroundColor: theme.color.white,
            labelStyle: { fontSize: 12 },
          }}>
          {tabs.isChatEnabled && (
            <Tab.Screen
              name="Chats"
              component={CometChatConversationListWithMessages}
            />
          )}
          {tabs.isUserListEnabled && (
            <Tab.Screen
              name="Users"
              component={CometChatUserListWithMessages}
            />
          )}
          {tabs.isGroupListEnabled && (
            <Tab.Screen
              name="Groups"
              component={CometChatGroupListWithMessages}
            />
          )}
          {tabs.isUserSettingsEnabled && (
            <Tab.Screen name="More" component={CometChatUserProfile} />
          )}
        </Tab.Navigator>
      ) : null}
    </CometChatContextProvider>
  );
}

export default CometChatUI;
