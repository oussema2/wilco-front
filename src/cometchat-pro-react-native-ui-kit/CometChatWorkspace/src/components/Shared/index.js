export { default as CometChatAvatar } from './CometChatAvatar';
export { default as CometChatSharedMedia } from './CometChatSharedMedia';
export { default as CometChatBadgeCount } from './CometChatBadgeCount';
export { default as CometChatBackdrop } from './CometChatBackdrop';
export { default as CometChatUserPresence } from './CometChatUserPresence';
