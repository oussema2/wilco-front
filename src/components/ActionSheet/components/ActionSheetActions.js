import React from 'react';
import { observer } from 'mobx-react';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { ActionSheetGradient } from '../../ActionSheetGradient';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetActions = () => {
	const actionSheetService = useActionSheetService();

	return (
		<ActionSheetGradient style={styles.actions}>
			{actionSheetService.actions.map( ( action ) => (
				<ActionSheetButton
					key={action.title}
					{...action}
					onPress={() => {
						actionSheetService.close();
						action.onPress();
					}}
				/>
			) )}
		</ActionSheetGradient>
	);
};

export default observer( ActionSheetActions );
