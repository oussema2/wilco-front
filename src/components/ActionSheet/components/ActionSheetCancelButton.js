import React from 'react';
import { observer } from 'mobx-react';

import { View } from 'react-native';
import { styles } from '../ActionSheet.styles';
import { ActionSheetButton } from '../../ActionSheetButton';
import { useActionSheetService } from '../../../providers/ActionSheetProvider';

const ActionSheetCancelButton = () => {
	const actionSheetService = useActionSheetService();

	return (
		<View style={styles.cancelButtonContainer}>
			<ActionSheetButton
				title="Cancel"
				onPress={() => actionSheetService.close()}
			/>
		</View>
	);
};

export default observer( ActionSheetCancelButton );
