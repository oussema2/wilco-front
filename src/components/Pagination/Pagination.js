import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { Pagination as NativePagination } from 'react-native-snap-carousel';
import { styles } from './Pagination.styles';

const Pagination = ( {
	testID, length, activeIndex
} ) => (
	<View testID={testID}>
		<NativePagination
			dotsLength={length}
			activeDotIndex={activeIndex}
			containerStyle={styles.container}
			dotContainerStyle={styles.dotContainer}
			dotStyle={styles.dot}
			inactiveDotStyle={styles.inactiveDot}
			inactiveDotOpacity={1}
			inactiveDotScale={1}
		/>
	</View>
);

Pagination.propTypes = {
	testID: PropTypes.string,
	length: PropTypes.number,
	activeIndex: PropTypes.number
};

Pagination.defaultProps = {
	testID: 'pagination-testID',
	length: 0,
	activeIndex: 0
};

export default Pagination;
