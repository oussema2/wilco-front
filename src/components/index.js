export { PrimaryButton } from './PrimaryButton';
export { TextInput } from './TextInput';
