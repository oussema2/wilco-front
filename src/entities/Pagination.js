import { makeAutoObservable } from 'mobx';

export default class Pagination {
	constructor( {
		page = 0,
		totalPages = -1,
		perPage = 15,
		isFirstPage = true,
		isLastPage = false
	} = {} ) {
		this.page = page;
		this.totalPages = totalPages;
		this.perPage = perPage;
		this.isFirstPage = isFirstPage;
		this.isLastPage = isLastPage;

		makeAutoObservable( this );
	}

	reset() {
		this.setPage( 0 );
		this.setIsLastPage( false );
		this.setTotalPages( -1 );
	}

	resetLastPage() {
		this.setPage( this.page - 1 );
		this.setIsLastPage( false );
	}

	setPage( page ) {
		this.page = page;
	}

	setTotalPages( totalPages ) {
		this.totalPages = totalPages;
	}

	setIsLastPage( isLastPage ) {
		this.isLastPage = isLastPage;
	}

	get hasMorePages( ) {
		return ( ( this.page < this.totalPages || this.totalPages === -1 )
				&& !this.isLastPage );
	}

	get nextPage() {
		return this.page + 1;
	}

	static fromJSON( jsonPagination ) {
		return new Pagination( {
			page: jsonPagination.current,
			totalPages: jsonPagination.pages,
			isFirstPage: jsonPagination.first_page,
			isLastPage: jsonPagination.last_page,
			perPage: jsonPagination.per_page
		} );
	}
}
