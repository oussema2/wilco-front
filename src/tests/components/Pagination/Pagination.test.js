import * as React from 'react';
import { render } from '@testing-library/react-native';
import { Pagination } from '../../../components/Pagination';

describe( 'Pagination', () => {
	describe( 'with the default props', () => {
		it( 'renders the Pagination component correctly', async () => {
			const { component, queryByTestId } = render(
				<Pagination />
			);
			expect( queryByTestId( 'pagination-testID' ) ).not.toBeNull();
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with activeIndex', () => {
		it( 'renders the Pagination component correctly', async () => {
			const { component } = render(
				<Pagination activeIndex={1} length={4} />
			);
			expect( component ).toMatchSnapshot();
		} );
	} );

	describe( 'with custom testID', () => {
		it( 'renders the Pagination component correctly', async () => {
			const { queryByTestId } = render(
				<Pagination testID="Custom-TestID" />
			);
			expect( queryByTestId( 'Custom-TestID' ) ).not.toBeNull();
		} );
	} );
} );
