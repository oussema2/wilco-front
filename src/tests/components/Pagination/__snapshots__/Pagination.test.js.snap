// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination with activeIndex renders the Pagination component correctly 1`] = `undefined`;

exports[`Pagination with the default props renders the Pagination component correctly 1`] = `undefined`;
